### PR TITLE
Created an acceptance tests

### PR DESCRIPTION
**Pull request by Veronichka**
Created an acceptance tests. So now we have following examples of Avro encoding/decoding:
- from version 1 to version 1
- from version 1 to version 2
- from version 1 to version 3 that is not compatible with version 1
- binary encoding/decoding from version 1 to version 1

### CODE REVIEW

**Comment by kurron** [comment]
It would be easier to understand the version tests if we used Semantic Versioning.  That way we would know what is supposed to be a compatible change versus a breaking change.

File: src/main/groovy/com/vherasymenko/avro/decoder/core/CourseInstallDecoderService.groovy [hunk 1]
```diff
@@ -21,11 +21,29 @@ class CourseInstallDecoderService implements CourseInstallDecoderPort {
     }
 
     @Override
-    void decodeCourseInstallEvent( String event, int schemaId ) {
+    void decodeCourseInstallV1ToV1( String event, int schemaId ) {
+        def schema = new Schema.Parser().parse( registryClient.fetch( schemaId ) )
+        log.info( 'The avro schema for event decoding : ' + schema.toString( true ) )
+
+        decoder.decodeEventWithOldSchema( schema, event )
```
Comment (kurron) [inline src/main/groovy/com/vherasymenko/avro/decoder/core/CourseInstallDecoderService.groovy:28]: Just curious about the name `decodeEventWithOldSchema`. Does this imply that the service needs to know ahead of time what version of an event it is dealing with and use the proper schema?

File: src/main/groovy/com/vherasymenko/avro/decoder/core/CourseInstallDecoderService.groovy [hunk 2]
```diff
@@ -21,11 +21,29 @@ class CourseInstallDecoderService implements CourseInstallDecoderPort {
...
         log.info( 'The avro schema for event decoding : ' + readSchema.toString( true ) )
 
-        decoder.decodeEvent( writeSchema, readSchema, event )
+        decoder.decodeEventWithNewSchema( writeSchema, readSchema, event )
+    }
+
+    @Override
+    void decodeCourseInstallV1ToV3( String event, int schemaId ) {
+        def writeSchema = new Schema.Parser().parse( registryClient.fetch( schemaId ) )
+        def readSchema = new Schema.Parser().parse( new File( 'src/main/resources/avro/course_install_3.avsc' ) )
```
Comment (kurron) [inline src/main/groovy/com/vherasymenko/avro/decoder/core/CourseInstallDecoderService.groovy:43]: I thought we were going to fetch **all** schema from the registry and not from the file system.  Am I wrong?

**Review by kurron — APPROVED** [review]
I thought we were also testing using generated Groovy files as well as the Map form.  I don't see any Gradle changes, which is where I would expect the code generator to be run.  Am I incorrect?

File: src/main/groovy/com/vherasymenko/avro/decoder/core/binary/AvroBinaryDecoderService.groovy [hunk 3]
```diff
@@ -7,6 +7,8 @@ import org.apache.avro.generic.GenericRecord
 import org.apache.avro.io.DatumReader
 import org.apache.avro.io.DecoderFactory
 
+import static com.vherasymenko.avro.schared.AvroConstants.DO_NOT_REUSE_STREAM
```
Comment (SergeyMormul) [inline src/main/groovy/com/vherasymenko/avro/decoder/core/binary/AvroBinaryDecoderService.groovy:10]: I'm pretty sure static import have to be before non-static import.

File: src/main/groovy/com/vherasymenko/avro/decoder/core/CourseInstallDecoderService.groovy (hunk 1 above)
Comment (Veronichka) [reply src/main/groovy/com/vherasymenko/avro/decoder/core/CourseInstallDecoderService.groovy:28]: Yes. We should now with what version of schema the event was encoded. We get the id of the schema in the message headers and retrieve it from the schema registry.

File: src/main/groovy/com/vherasymenko/avro/decoder/core/CourseInstallDecoderService.groovy (hunk 2 above)
Comment (Veronichka) [reply src/main/groovy/com/vherasymenko/avro/decoder/core/CourseInstallDecoderService.groovy:43]: I plan to do this in the next PR. I think it's too much to include that functionality to this one.

**Review by SergeyMormul — APPROVED** [review]

File: src/main/groovy/com/vherasymenko/avro/decoder/core/binary/AvroBinaryDecoderService.groovy (hunk 3 above)
Comment (Veronichka) [reply src/main/groovy/com/vherasymenko/avro/decoder/core/binary/AvroBinaryDecoderService.groovy:10]: It depends on your IDE settings. I didn't add this import by hand.

**Comment by Veronichka** [comment]
The code generation is done by the `gradle-avro-plugin`. That example is already in the `master` branch.

**Review by clefevre — APPROVED** [review]

**Comment by clefevre** [comment]
The fact that we aren't using semantic versioning definitely hurt when looking at versions 1 to 2 as being non-breaking, yet 3 is breaking. But not something I would heavily redesign the proof of concept over.

**Review by AnnaKhoma — APPROVED** [review]

**Review by kulminsky — APPROVED** [review]

**Pull request merged**